File: src/scaffold/gitignore.ts
import fs from 'node:fs'
import path from 'node:path'

const GITIGNORE_ENTRIES = [
  '.ai/memory/',
  '.env',
  '.env.local',
  '.env*.local',
]

export function checkGitignoreGuidance(targetDir: string): void {
  const gitignorePath = path.join(targetDir, '.gitignore')

  if (!fs.existsSync(gitignorePath)) {
    console.log('\n💡 Consider creating a .gitignore with:')
    for (const entry of GITIGNORE_ENTRIES) {
      console.log(`   ${entry}`)
    }
    return
  }

  const content = fs.readFileSync(gitignorePath, 'utf-8')
  const missing: string[] = []

  for (const entry of GITIGNORE_ENTRIES) {
    // Check if the entry (or a broader pattern covering it) is already present
    const searchPattern = entry.replace(/\*/g, '').replace(/\./g, '\\.')
    if (!content.includes(entry) && !new RegExp(searchPattern).test(content)) {
      missing.push(entry)
    }
  }

  if (missing.length > 0) {
    console.log('\n💡 Consider adding to .gitignore:')
    for (const entry of missing) {
      console.log(`   ${entry}`)
    }
  }
}

File: internal/scaffold/gitignore.go
package scaffold

import (
	"fmt"
	"os"
	"path/filepath"
	"regexp"
	"strings"
)

// GitignoreEntries lists the recommended .gitignore entries for ai-setup.
var GitignoreEntries = []string{
	".ai/memory/",
	".env",
	".env.local",
	".env*.local",
}

// CheckGitignoreGuidance checks if .gitignore has recommended entries and
// prints guidance for missing ones. Ported from src/scaffold/gitignore.ts.
func CheckGitignoreGuidance(targetDir string) {
	gitignorePath := filepath.Join(targetDir, ".gitignore")

	data, err := os.ReadFile(gitignorePath)
	if err != nil {
		if os.IsNotExist(err) {
			fmt.Println("\n💡 Consider creating a .gitignore with:")
			for _, entry := range GitignoreEntries {
				fmt.Printf("   %s\n", entry)
			}
			return
		}
		return
	}

	content := string(data)
	var missing []string

	for _, entry := range GitignoreEntries {
		searchPattern := strings.ReplaceAll(entry, "*", "")
		searchPattern = regexp.QuoteMeta(searchPattern)
		matched, _ := regexp.MatchString(searchPattern, content)
		if !strings.Contains(content, entry) && !matched {
			missing = append(missing, entry)
		}
	}

	if len(missing) > 0 {
		fmt.Println("\n💡 Consider adding to .gitignore:")
		for _, entry := range missing {
			fmt.Printf("   %s\n", entry)
		}
	}
}
